### ISSUE TITLE
[yamahareceiver] WXA-10 or WXA-30 Yamaha Musiccast devices not supported

### ISSUE THREAD

**Issue by dastrix80** [openhab/openhab-addons#4370]
Hi Guys

Can these devices please be added into support? Im running 2.4 of the binding

`233 │ Active   │  80 │ 2.4.0.201808271627     │ YamahaReceiver Binding`

I get this error:

```
09:02:32.398 [INFO ] [smarthome.event.InboxRemovedEvent    ] - Discovery Result with UID 'yamahareceiver:yamahaAV:9ab0c000_f668_11de_9976_00a0def90b3f' has been removed.
09:02:33.374 [INFO ] [smarthome.event.ItemStateChangedEvent] - C3945_CPUTemp changed from 58 to 59
09:02:35.652 [INFO ] [ig.discovery.internal.PersistentInbox] - Added new thing 'yamahareceiver:yamahaAV:9ab0c000_f668_11de_9976_00a0def90b3f' to inbox.
09:02:35.652 [INFO ] [smarthome.event.InboxAddedEvent      ] - Discovery Result with UID 'yamahareceiver:yamahaAV:9ab0c000_f668_11de_9976_00a0def90b3f' has been added.
09:02:35.850 [INFO ] [smarthome.event.ItemStateChangedEvent] - OpenHAB_Network_DataRecevied changed from 411 to 412
09:02:35.854 [INFO ] [smarthome.event.ItemStateChangedEvent] - OpenHAB_Network_DataSent changed from 174 to 177
09:02:35.856 [INFO ] [smarthome.event.ItemStateChangedEvent] - OpenHAB_Cpu_Load changed from 2.0 to 2.3
09:02:38.138 [INFO ] [smarthome.event.ItemStateChangedEvent] - Kris_MobileLastSeen changed from 2018-12-15T09:01:27.000+1100 to 2018-12-15T09:02:20.000+1100
09:02:38.145 [INFO ] [ipse.smarthome.model.script.Blue Iris] - Selective motion recording ON because someone came home and its day
09:02:38.145 [INFO ] [ipse.smarthome.model.script.Blue Iris] - Selective motion recording ON because someone came home and its day
09:02:42.618 [INFO ] [smarthome.event.InboxRemovedEvent    ] - Discovery Result with UID 'yamahareceiver:yamahaAV:9ab0c000_f668_11de_9976_00a0def90b3f' has been removed.
09:02:42.621 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - 'yamahareceiver:yamahaAV:9ab0c000_f668_11de_9976_00a0def90b3f' changed from UNINITIALIZED to INITIALIZING
09:02:42.629 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - 'yamahareceiver:yamahaAV:9ab0c000_f668_11de_9976_00a0def90b3f' changed from INITIALIZING to OFFLINE (CONFIGURATION_PENDING): Waiting for data
09:02:42.640 [WARN ] [areceiver.handler.YamahaBridgeHandler] - Communication error. Please review your Yamaha thing configuration.
java.io.IOException: Changing a value on the Yamaha AVR failed: <?xml version="1.0" encoding="utf-8"?><YAMAHA_AV cmd="GET"><System><Config>GetParam</Config></System></YAMAHA_AV>
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLConnection.postMessage(XMLConnection.java:83) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLConnection.sendReceive(XMLConnection.java:115) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolService.getResponse(XMLProtocolService.java:77) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.DeviceInformationXML.update(DeviceInformationXML.java:87) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.handler.YamahaBridgeHandler.updateDeviceInformation(YamahaBridgeHandler.java:240) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.handler.YamahaBridgeHandler.connectionEstablished(YamahaBridgeHandler.java:327) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolFactory.createConnection(XMLProtocolFactory.java:26) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.handler.YamahaBridgeHandler.initialize(YamahaBridgeHandler.java:307) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at sun.reflect.GeneratedMethodAccessor77.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:153) [94:org.eclipse.smarthome.core:0.10.0.201808242044]
        at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53) [94:org.eclipse.smarthome.core:0.10.0.201808242044]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
09:02:42.650 [INFO ] [ome.event.ThingStatusInfoChangedEvent] - 'yamahareceiver:yamahaAV:9ab0c000_f668_11de_9976_00a0def90b3f' changed from OFFLINE (CONFIGURATION_PENDING): Waiting for data to OFFLINE (COMMUNICATION_ERROR): Changing a value on the Yamaha AVR failed: <?xml version="1.0" encoding="utf-8"?><YAMAHA_AV cmd="GET"><System><Config>GetParam</Config></System></YAMAHA_AV>
09:02:42.790 [ERROR] [rnal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.yamahareceiver.handler.YamahaBridgeHandler@556dbfcb': Could not communicate with the device
java.lang.RuntimeException: Could not communicate with the device
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.InputConverterXML.<init>(InputConverterXML.java:58) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolFactory.InputConverter(XMLProtocolFactory.java:115) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.handler.YamahaBridgeHandler.initialize(YamahaBridgeHandler.java:308) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at sun.reflect.GeneratedMethodAccessor77.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:153) [94:org.eclipse.smarthome.core:0.10.0.201808242044]
        at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53) [94:org.eclipse.smarthome.core:0.10.0.201808242044]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.io.IOException: Changing a value on the Yamaha AVR failed: <?xml version="1.0" encoding="utf-8"?><YAMAHA_AV cmd="GET"><Main_Zone><Input><Input_Sel_Item>GetParam</Input_Sel_Item></Input></Main_Zone></YAMAHA_AV>
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLConnection.postMessage(XMLConnection.java:83) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLConnection.sendReceive(XMLConnection.java:115) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolService.getResponse(XMLProtocolService.java:77) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolService.getZoneResponse(XMLProtocolService.java:65) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolService.getInputs(XMLProtocolService.java:97) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.InputConverterXML.createInputsWithoutMapping(InputConverterXML.java:96) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.InputConverterXML.<init>(InputConverterXML.java:55) ~[?:?]
        ... 11 more
09:02:42.809 [ERROR] [home.core.thing.internal.ThingManager] - Exception occurred while initializing handler of thing 'yamahareceiver:yamahaAV:9ab0c000_f668_11de_9976_00a0def90b3f': Could not communicate with the device
java.lang.RuntimeException: Could not communicate with the device
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.InputConverterXML.<init>(InputConverterXML.java:58) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolFactory.InputConverter(XMLProtocolFactory.java:115) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at org.openhab.binding.yamahareceiver.handler.YamahaBridgeHandler.initialize(YamahaBridgeHandler.java:308) [233:org.openhab.binding.yamahareceiver:2.4.0.201808271627]
        at sun.reflect.GeneratedMethodAccessor77.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:153) [94:org.eclipse.smarthome.core:0.10.0.201808242044]
        at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53) [94:org.eclipse.smarthome.core:0.10.0.201808242044]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.io.IOException: Changing a value on the Yamaha AVR failed: <?xml version="1.0" encoding="utf-8"?><YAMAHA_AV cmd="GET"><Main_Zone><Input><Input_Sel_Item>GetParam</Input_Sel_Item></Input></Main_Zone></YAMAHA_AV>
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLConnection.postMessage(XMLConnection.java:83) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLConnection.sendReceive(XMLConnection.java:115) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolService.getResponse(XMLProtocolService.java:77) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolService.getZoneResponse(XMLProtocolService.java:65) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.XMLProtocolService.getInputs(XMLProtocolService.java:97) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.InputConverterXML.createInputsWithoutMapping(InputConverterXML.java:96) ~[?:?]
        at org.openhab.binding.yamahareceiver.internal.protocol.xml.InputConverterXML.<init>(InputConverterXML.java:55) ~[?:?]
        ... 11 more
```

Thank you

**Comment by zarusz** [comment]
Hi, 

The YamahaReceiver binding does not support the Yamaha MusicCast protocol (aka Yamaha Extended Control) that is required to communicate with these devices. This is by design and not a bug. There is a related [long community topic](https://community.openhab.org/t/yamaha-musiccast-binding/15060/125). There was some [work done by other community members](https://community.openhab.org/t/yamaha-musiccast-binding-revival/57833/4) that might be of help. Due to lack of time, I do not currently plan to add MusicCast support (... as much as I would love to).

**Comment by dastrix80** [comment]
Thanks Thomas

**Comment by coop-git** [comment]
Please have a look again at the [community topic](https://community.openhab.org/t/yamaha-musiccast-binding-revival/57833/115)
A few weeks ago I started working on a new binding. Basic functionality is available, first draft of MusicCast linking is builtin as well. All feedback is welcome.

Goal is to have it included in OH3, standalone JAR is/will be available for OH 2.5.x
Current development of binding is available in [this repo](https://github.com/coop-git/YamahaMusicCast)

**Comment by hmerk** [comment]
Can this one be closed as PR #11880 was merged and the binding is included in openHAB 3.3 release.
